tests/integration_tests/functional/test_api.py

```diff
@@ -319,6 +298,101 @@ def test_net_api_put_update_pre_boot(test_microvm_with_api):
     assert test_microvm.api_session.is_status_no_content(response.status_code)
 
 
+def test_api_mmds_config(test_microvm_with_api):
```

Comment:
What happens when an interface is configured via /network-interfaces, mmds is enabled on it, and then the interface deleted from /network-interfaces?

Reply:
```rust
        // Create `MmdsNetworkStack` and configure the IPv4 address for
        // existing built network devices whose names are defined in the
        // network interface ID list.
        for net_device in self.net_builder.iter_mut() {
            let mut net_device_lock = net_device.lock().expect("Poisoned lock");
            if network_interfaces.contains(net_device_lock.id()) {
                net_device_lock.configure_mmds_network_stack(ipv4_addr);
            } else {
                net_device_lock.disable_mmds_network_stack();
            }
        }
```

This piece of code from `set_mmds_config()` handles this case. We iterate through all network devices attached and if the ID of the net iface is present in the `network_interfaces` list provided through `MmdsConfig`, then we configure it to allow MMDS requests. Otherwise, we disable the MMDS Network Stack from that interface. This ensures that only the interfaces specified in the list accept forwarding requests to MMDS.

Reply:
I think Radu was referring to the case where an user appends an interface and then deletes it. We do not offer a delete api endpoint in Firecracker. Also, the patch request on network interface allows updating of the rate limiters only.